rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -55,31 +59,63 @@ SequentialWriter::SequentialWriter(
   storage_(nullptr),
   metadata_io_(std::move(metadata_io)),
   converter_(nullptr),
+  compressor_{nullptr},
   max_bagfile_size_(rosbag2_storage::storage_interfaces::MAX_BAGFILE_SIZE_NO_SPLIT),
   topics_names_to_info_(),
-  metadata_()
-{}
+  metadata_(),
+  compression_mode_{CompressionMode::NONE} {}
+
 
 SequentialWriter::~SequentialWriter()
 {
   reset();
 }
 
-void SequentialWriter::init_metadata()
+void SequentialWriter::check_bagfile_size(const StorageOptions & storage_options)
+{
+  if (storage_options.max_bagfile_size != 0 &&
+    storage_options.max_bagfile_size < storage_->get_minimum_split_file_size())
+  {
+    throw std::invalid_argument{
+            "Invalid bag splitting size given. Please provide a different value."};
+  }
+}
```

Comment:
Is this relevant to this PR?
I think this code block could be extracted and done in a separate PR.